File: win-prog/ProgWinCS/Images and Bitmaps/DualWink/DualWink.cs
//---------------------------------------
// DualWink.cs © 2001 by Charles Petzold
//---------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class DualWink: Wink
{
     Image[] aimageRev = new Image[4];

     public new static void Main()
     {
          Application.Run(new DualWink());
     }
     public DualWink()
     {
          Text = "Dual " + Text;

          for(int i = 0; i < 4; i++)
          {
               aimageRev[i] = (Image) aimage[i].Clone();
               aimageRev[i].RotateFlip(RotateFlipType.RotateNoneFlipX);
          }
     }
     protected override void TimerOnTick(object obj, EventArgs ea)
     {
          Graphics grfx = CreateGraphics();

          grfx.DrawImage(aimage[iImage], 
                    ClientSize.Width / 2,
                    (ClientSize.Height - aimage[iImage].Height) / 2,
					aimage[iImage].Width, aimage[iImage].Height);

          grfx.DrawImage(aimageRev[3 - iImage],
                    ClientSize.Width / 2 - aimageRev[3 - iImage].Width,
                    (ClientSize.Height - aimageRev[3 - iImage].Height) / 2,
					aimageRev[3 - iImage].Width,
					aimageRev[3 - iImage].Height);

          grfx.Dispose();

          iImage += iIncr;

          if (iImage == 3)
               iIncr = -1;
          
          else if (iImage == 0)
               iIncr = 1;
     }
}

File: win-prog/ProgWinVB/Images and Bitmaps/DualWink/DualWink.vb
'-----------------------------------------
' DualWink.vb (c) 2002 by Charles Petzold
'-----------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class DualWink
    Inherits Wink

    Private aimgRev(3) As Image

    Shared Shadows Sub Main()
        Application.Run(New DualWink())
    End Sub

    Sub New()
        Text = "Dual " & Text

        Dim i As Integer
        For i = 0 To 3
            aimgRev(i) = DirectCast(aimg(i).Clone(), Image)
            aimgRev(i).RotateFlip(RotateFlipType.RotateNoneFlipX)
        Next i
    End Sub

    Protected Overrides Sub TimerOnTick(ByVal obj As Object, _
                                        ByVal ea As EventArgs)
        Dim grfx As Graphics = CreateGraphics()

        grfx.DrawImage(aimg(iImage), _
                ClientSize.Width \ 2, _
                (ClientSize.Height - aimg(iImage).Height) \ 2, _
                aimg(iImage).Width, aimg(iImage).Height)

        grfx.DrawImage(aimgRev(3 - iImage), _
                ClientSize.Width \ 2 - aimgRev(3 - iImage).Width, _
                (ClientSize.Height - aimgRev(3 - iImage).Height) \ 2, _
                aimgRev(3 - iImage).Width, _
                aimgRev(3 - iImage).Height)
        grfx.Dispose()

        iImage += iIncr
        If iImage = 3 Then
            iIncr = -1
        ElseIf iImage = 0 Then
            iIncr = 1
        End If
    End Sub
End Class
